lib/raidenclient/RaidenClient.ts

```diff
@@ -272,14 +278,17 @@ class RaidenClient extends SwapClient {
   }
 
   /**
-   * Returns the total balance available across all channels.
+   * Returns the total balance available across all channels for a specified currency.
    */
-  public channelBalance = async (): Promise<ChannelBalance> => {
-    // TODO: refine logic to determine balance per token rather than all combined
-    const channels = await this.getChannels();
+  public channelBalance = async (currency?: string): Promise<ChannelBalance> => {
+    if (!currency) {
+      return { balance: 0, pendingOpenBalance: 0 };
+    }
+
+    const channels = await this.getChannels(this.tokenAddresses.get(currency));
     const balance = channels.filter(channel => channel.state === 'opened')
       .map(channel => channel.balance)
-      .reduce((acc, sum) => sum + acc, 0);
+      .reduce((acc, sum) => sum + (acc / RaidenClient.UNITS_PER_CURRENCY[currency]), 0);
```

Comment:
Think we want to switch places with `sum` and `acc` here (and rename them to `sum` and `channelBalance`. 
`arr.reduce(callback(accumulator, currentValue[, index[, array]]), [, initialValue])`

Maybe do the conversion after reduce so that it's only going to be calculated once?

I think this will cause confusion on the CLI level if we show satoshis per coin as it's meant to be our internal unit only. What do you think about converting Raiden channel balance values to `satsToCoinsStr` and then doing additional conversion to smallest unit of the currency on CLI level?

Reply:
I'm a bit confused by what you mean in the first paragraph, could you clarify? I went ahead and made the conversion happen after reduce though.

We can have the cli print whole units for the "formatted" output, that'd be consistent with how we're displaying amounts most other places. That should be a separate PR though I think.

Reply:
What I was trying to say was that the first argument for the callback of reduce is `sum` and the second `acc`.

With the previous logic:
```
.reduce((acc, sum) => {
  console.log('sum is', sum, 'and converted', (acc / RaidenClient.UNITS_PER_CURRENCY[currency]));
  return sum + (acc / RaidenClient.UNITS_PER_CURRENCY[currency]);
}, 0);
```

Printed:
```
[1] sum is 75000000000000000 and converted 0
```

As a result converting back to satoshis per unit didn't have any effect:
```
➜  xud git:(channel-mgmt-plus-raiden-balance) ✗ ./bin/xucli channelbalance
{
  "balancesMap": [
    ...
    [
      "WETH",
      {
        "balance": 75000000000000000,
        "pendingOpenBalance": 0
      }
    ]
  ]
}
```

With the new logic this isn't an issue, but I'd still rename the arguments.

As for converting back to smallest units on the CLI level - let's open an issue for that to discuss. I think it will cause confusion for the end user:
```
"WETH",
{
  "balance": 7500000,
  "pendingOpenBalance": 0
}
```

Reply:
Gotcha.